cognite/neat/_graph/transformers/_base.py

```diff
@@ -12,3 +16,119 @@ class BaseTransformer(ABC):
     @abstractmethod
     def transform(self, graph: Graph) -> None:
         raise NotImplementedError()
+
+
+class BaseTransformerStandardised(ABC):
+    """Standardised base transformer to use in case a transformer is adding or removing triples from a graph. If you
+    are doing more specialised operations, please overwrite the .transform() method.
+    """
+
+    _use_only_once: bool = False
+    _need_changes: ClassVar[frozenset[str]] = frozenset()
+    _use_iterate_bar_threshold: int = 500
+
+    @abstractmethod
+    def description(self) -> str:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def operation(self) -> Literal["add", "remove"]:
```

Comment:
If more complicated operations than add or remove, create specialised transformation in my opinion